subgraph/src/public-lock.ts

```diff
@@ -440,11 +441,17 @@ export function createReceipt(event: ethereum.Event): void {
       // including one for the ERC20 transfer
       for (let i = 0; i < logs.length; i++) {
         const txLog = logs[i]
-        if (
-          txLog.address == tokenAddress &&
```

Comment:
`txLog.address == tokenAddress` this check should not be right looking at this logs we have 

**address**: 0x38181253e065b18df0006644f5fa9c2fda46cc75
**hashed signature**: 0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef

https://goerli.etherscan.io/tx/0xaa9a7112a5a2738e6d45b3966f2df358d25c535fa53441639cf2ab5ecedad4ed#eventlog

Reply:
@kalidiagne indeed, this transaction is NOT on an ERC20 lock... so the `payer` should be the sender of the transaction... so the problem is not about the `Transfer` event!